src/hotspot/share/runtime/thread.cpp

```diff
@@ -564,43 +564,32 @@ void Thread::send_async_exception(oop java_thread, oop java_throwable) {
 
 class TraceSuspendDebugBits : public StackObj {
  private:
-  JavaThread * jt;
-  bool         is_wait;
-  bool         called_by_wait;  // meaningful when !is_wait
-  uint32_t *   bits;
+  JavaThread* _jt;
+  uint32_t*   _bits;
 
  public:
-  TraceSuspendDebugBits(JavaThread *_jt, bool _is_wait, bool _called_by_wait,
-                        uint32_t *_bits) {
-    jt             = _jt;
-    is_wait        = _is_wait;
-    called_by_wait = _called_by_wait;
-    bits           = _bits;
+  TraceSuspendDebugBits(JavaThread *jt, uint32_t *bits) : _jt(jt), _bits(bits) {
   }
 
   ~TraceSuspendDebugBits() {
-    if (!is_wait) {
 #if 1
-      // By default, don't trace bits for is_ext_suspend_completed() calls.
-      // That trace is very chatty.
-      return;
+    // By default, don't trace bits for is_ext_suspend_completed() calls.
+    // That trace is very chatty.
+    return;
 #else
```

Comment:
Without the !is_wait check none of the code below line 583 is reachable now. I would remove this now.

Reply:
Since only the destructor contains any actual functionality, which is now unreachable, should I remove the entire TraceSuspendDebugBits?